overlord/state/state.go

```diff
@@ -566,5 +566,7 @@ func ReadState(backend Backend, r io.Reader) (*State, error) {
 	s.modified = false
 	s.cache = make(map[interface{}]interface{})
 	s.pendingChangeByAttr = make(map[string]func(*Change) bool)
+	s.changeHandlers = make(map[int]func(chg *Change, old Status, new Status))
+	s.taskHandlers = make(map[int]func(t *Task, old Status, new Status))
```

Comment:
I wonder if there is a way to extend TestEmptyStateDataAndCheckpointReadAndSet to cover this code more

Reply:
I extended the test and added a reflection test that detects if any maps have not been initialized after the ReadState call. I verified this by removing one of the `make(...)` calls in ReadState and the test now correctly fails.